src/power_grid_model_io/converters/pandapower_converter.py

```diff
@@ -0,0 +1,361 @@
+        pgm_lines["to_status"] = pp_lines["in_service"] & pp_to_switches["closed"]
+        pgm_lines["r1"] = pp_lines["r_ohm_per_km"] * pp_lines["length_km"]
+        pgm_lines["x1"] = pp_lines["x_ohm_per_km"] * pp_lines["length_km"]
+        pgm_lines["c1"] = pp_lines["c_nf_per_km"] * pp_lines["length_km"]
+        pgm_lines["tan1"] = 0.0
+        pgm_lines["i_n"] = pp_lines["max_i_ka"] * 1e3
+
+        self.pgm_data["line"] = pgm_lines
+
+    def _create_pgm_input_sources(self):
+        assert "source" not in self.pgm_data
+
+        pp_ext_grid = self.pp_data["ext_grid"]
+
+        pgm_sources = initialize_array(data_type="input", component_type="source", shape=len(pp_ext_grid))
+        pgm_sources["id"] = self._generate_ids("source", pp_ext_grid.index)
+        pgm_sources["node"] = self._get_ids("bus", pp_ext_grid["bus"])
+        pgm_sources["status"] = pp_ext_grid["in_service"]
+        pgm_sources["u_ref"] = pp_ext_grid["vm_pu"]
+
+        self.pgm_data["source"] = pgm_sources
+
+    def _create_pgm_input_shunts(self):
+        assert "shunt" not in self.pgm_data
+
+        pp_shunt = self.pp_data["shunt"]
+
+        pgm_shunts = initialize_array(data_type="input", component_type="shunt", shape=len(pp_shunt))
+        pgm_shunts["id"] = self._generate_ids("shunt", pp_shunt.index)
+        pgm_shunts["node"] = self._get_ids("bus", pp_shunt["bus"])
+        pgm_shunts["status"] = pp_shunt["in_service"]
+        pgm_shunts["g1"] = (pp_shunt["p_mw"] * 1e6) / ((pp_shunt["vn_kv"] * 1e3) * (pp_shunt["vn_kv"] * 1e3))
+        pgm_shunts["b1"] = (pp_shunt["q_mvar"] * 1e6) / ((pp_shunt["vn_kv"] * 1e3) * (pp_shunt["vn_kv"] * 1e3))
+
+        self.pgm_data["shunt"] = pgm_shunts
+
+    def _create_pgm_input_sym_gens(self):
+        assert "sym_gen" not in self.pgm_data
+
+        pp_sgens = self.pp_data["sgen"]
+
+        pgm_sym_gens = initialize_array(data_type="input", component_type="sym_gen", shape=len(pp_sgens))
+        pgm_sym_gens["id"] = self._generate_ids("sym_gen", pp_sgens.index)
+        pgm_sym_gens["node"] = self._get_ids("bus", pp_sgens["bus"])
+        pgm_sym_gens["status"] = pp_sgens["in_service"]
+        pgm_sym_gens["p_specified"] = pp_sgens["p_mw"] * 1e6
+        pgm_sym_gens["q_specified"] = pp_sgens["q_mvar"] * 1e6
+
+        self.pgm_data["sym_gen"] = pgm_sym_gens
+
+    def _create_pgm_input_sym_loads(self):
+        assert "sym_load" not in self.pgm_data
+
+        pp_loads = self.pp_data["load"]
+
+        n_loads = len(pp_loads)
+
+        pgm_sym_loads = initialize_array(data_type="input", component_type="sym_load", shape=3 * n_loads)
+
+        const_i_multiplier = pp_loads["const_i_percent"] * (1e-2 * 1e6)
+        const_z_multiplier = pp_loads["const_z_percent"] * (1e-2 * 1e6)
+        const_p_multiplier = 1e6 - const_i_multiplier - const_z_multiplier
+
+        pgm_sym_loads["id"][:n_loads] = self._generate_ids("sym_load_const_power", pp_loads.index)
```

Comment:
Something more must be thought in generating ids of sym_load for recombining results

Reply:
Could you explain the issue or maybe give an idea on how to solve it?

Reply:
Just that it needs to be thought of while making output conversions. I think you have it covered here.